### ISSUE TITLE
Add standardized weight colors and display to Polygon2D weight editor

### ISSUE THREAD

**Issue by CantyCanadian** [godotengine/godot-proposals#15158]
### Problem or limitation

The method currently used to show the weights given to vertices in the Polygon2D editor is very imprecise and lacks accessibility for visually-impaired users. In the code, we assign a color between black and white using the weight as the ratio, which makes the difference between small values (say, 0 and 0.01) nearly impossible to differentiate despite it resulting in a significant change in your mesh.

### Proposed improvement

The proposed solution is twofold:

1. Use a similar system of color-to-weight display as similar applications. 

For example, 
[Blender](https://docs.blender.org/manual/en/latest/sculpt_paint/weight_paint/introduction.html) uses "blue = 0, cyan = 0.25, green = 0.5, yellow/orange = 0.75, red = 1";
[3DS Max](https://download.autodesk.com/us/3dsmax/2012help/index.html?url=files/GUID-C5EB44E2-5A3E-44DF-8B64-A0E00B8EDC7-475.htm,topicNumber=d28e101076) uses a slightly different scale "blue = 0, green = 0.25, yellow = 0.5, orange = 0.75, red = 1".

Using such a scale means more meaningful differences between smaller values and makes it much easier for someone who is visually-impaired to see said differences.

2. Allow the user to see the weights with a hotkey. For example, if holding 'alt', each vertices has a label that appears showing the currently assigned weight.

### Proposal review

- [x] Yes, this proposal is highly specific and actionable.
